pandas/core/indexes/datetimes.py

```diff
@@ -546,6 +581,9 @@ def _fast_union(self, other):
 
     def _wrap_setop_result(self, other, result):
         name = get_op_result_name(self, other)
+        if isinstance(result, list):
```

Comment:
is_list_like ? what is this for?

Reply:
I don't recall. It seems to be not hit by the tests. I'll try removing it.